core/lib/src/rocket.rs

```diff
@@ -842,41 +928,40 @@ impl Rocket {
         server.await
     }
 
-    /// Starts the application server and begins listening for and dispatching
-    /// requests to mounted routes and catchers. This function does not return
-    /// unless a shutdown is requested via a [`ShutdownHandle`] or there is an
-    /// error.
-    ///
-    /// This is a convenience function that creates a suitable default runtime
-    /// and launches the server on that runtime. If you already have a runtime,
-    /// use the [`Rocket::serve`] method instead.
-    ///
-    /// # Error
+    pub(crate) fn take_inner(&mut self) -> RocketInner {
```

Comment:
Can we make this safer? Ideally we'd like to make it easy to prove that `self.inner` is "always" `Some` after some logically atomic operation. I _think_ we'd like something to the effect of:

```rust
fn with_inner(&mut self, f: impl FnOnce(RocketInner) -> RocketInner) {
    let inner = self.inner.take().expect("...");
    self.inner = Some(f(inner));
}
```

Reply:
I believe some callers can and some callers can't handle that signature. I will see about making those more clear.